packages/types/src/codec/utils/encodeTypes.ts

```diff
@@ -156,10 +156,10 @@ function encodeType (typeDef: Pick<TypeDef, any>): string {
   return encoder(typeDef as TypeDef);
 }
 
-export function encodeTypeDef (typeDef: Pick<TypeDef, any>): string {
+export function encodeTypeDef (typeDef: Pick<TypeDef, any>, asRaw?: boolean): string {
```

Comment:
As discussed, revert this, the issue actually lies elsewhere - the "type" in contracts should map to the raw type.